analytic_billing_plan/models/analytic_billing_plan.py

```diff
@@ -109,16 +69,33 @@ def action_button_draft(self):
                       'with a paid invoice.'))
             for line in rec.analytic_billing_plan_line_ids:
                 line.move_id.unlink()
-                line.write(
-                    {
-                        'state': 'draft',
-                    })
+                line.write({'state': 'draft', })
             rec.write({'state': 'draft'})
             rec.message_post(
                 _('<strong>Billing Request Drafted.</strong><ul>'
                   '<li><strong>Confirmed by: </strong>%s</li>'
                   '<li><strong>Confirmed at: </strong>%s</li>'
                   '</ul>') % (self.env.user.name, fields.Datetime.now()))
+    
+    @api.model
+    def _prepare_invoice(self):
```

Comment:
This method should only returns de dict and should not be a private method please remove the initial "_"
 :)
Something like this:
return {
  'project_id': project.id,
  'partner_id': partner,
}

and mut be called by self.env['account.invoice'].create(self.prepare_invoice())